Enumerables.rb

```diff
@@ -0,0 +1,70 @@
+module Enumerable
+    def my_each
+        i = 0
+        while i<self.size
+            yield self[i]
+            i+=1
+        end
+    end
+
+    def my_each_with_indeitem
+        i = 0
+        result = []
+        while i < self.size
+            result << yield(self[i],i)
+            i +=1
+        end
+        result
+    end
+
+    def my_select
+        result=[]
+        my_each {|item| result<<item if yield item}
+        result
+    end
+
+    def my_all?
+        my_each{|item| return false unless yield item}
+        true
+    end
+
+    def my_any?
+        my_each{|item| return true if yield item}
+        false
+    end
+
+    def my_none
+        my_each{|item| return false if yield item}
+        true
+    end
+
+    def my_count(arg=nil)
+        count=0
+        my_each{|item| count += 1 if yield item} if block_given? and arg.nil?
+        my_each{|item| count += 1 if arg == item} if !block_given? and !arg.nil?
+        my_each{|item| count += 1} if !block_given? and arg.nil?
+        count
+    end
+
+    def my_map(&block)
+        result = []
+        my_each{|item| result<<block.call(item)}
+        result
+    end
+
+    def my_inject(init=nil)
```

Comment:
- [ ] `my_inject` will always return zero if applied a multiplication operation on it as the initial value of result is zero so multipled by any other number will be always zero.